code/datums/status_effects/xeno_buffs.dm

```diff
@@ -518,7 +518,7 @@
 
 		if(do_after(owner_xeno, KNOCKDOWN_DURATION, IGNORE_HELD_ITEM, target))
 			owner_xeno.gain_plasma(plasma_gain_on_hit)
-
+			SEND_SIGNAL(target, COMSIG_XENO_GREENBLOOD_DRAIN, owner_xeno.xeno_caste.drain_plasma_gain, owner_xeno)
```

Comment:
2 issues
1) this signal is sent in 2 places which is not how signals should be used 
2) you should rename so they're generic events, signals should be "x thing is happening to anyone who cares" not " do x thing"

Reply:
should be good now.